server/src/routines/background/chat.ts

```diff
@@ -23,55 +16,17 @@ function* chatMessageSaga(game: GameModel, action: AnyAction) {
 	if (message.length < 1) return
 	if (message.length > 140) return
 
-	const hermitNames = Object.values(HERMIT_CARDS).map((card) => {
-		return {shortName: card.name.toLowerCase().replace(' ', '_'), longName: card.id.split('_')[0]}
-	})
-
-	var splitMessage: Array<MessageChunk> = [
-		{
-			message: message,
-			type: 'text',
-		},
-	]
-
-	hermitNames.forEach((name) => {
-		splitMessage = splitMessage.flatMap((chunk) => {
-			if (chunk.type === 'emoji') return chunk
-			const splitString = chunk.message.split(new RegExp('(:' + name.shortName + ':)', 'g'))
-			const formattedSplitString: Array<MessageChunk> = []
-			splitString.forEach((chunk) => {
-				if (chunk === '') return
-				const colonlessChunk = chunk.substring(1, chunk.length - 1)
-				formattedSplitString.push({
-					message: colonlessChunk === name.shortName ? name.longName : chunk,
-					type: colonlessChunk === name.shortName ? 'emoji' : 'text',
-				})
-			})
-			return formattedSplitString
-		})
-	})
-
-	const messageArray = splitMessage.map((chunk) => {
-		if (chunk.type === 'emoji') {
-			return {
-				text: `images/hermits-emoji/${chunk.message}.png`,
-				censoredText: '',
-				alt: chunk.message,
-				format: 'image',
-			}
-		}
-		return {
-			text: chunk.message,
-			censoredText: profanityFilter(chunk.message),
-			format: 'plain',
-		}
-	})
-
 	game.chat.push({
+		message: concatFormattedTextNodes(
+			formatText(`$p${game.players[playerId].name}$ `),
```

Comment:
Player names are not tested for profanity
```suggestion
			formatText(`$p${game.players[playerId].name}$ `, {censor: true}),
```

Reply:
I agree with adding this! I left it out before but it does make sense considering past history!

Reply:
Yeah. Just make sure you are also censoring player names in all the other places they are displayed if you make this change.